atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+            TableReference tableRef, Iterable<RangeRequest> rangeRequests, long timestamp) {
+        Map<RangeRequest, TokenBackedBasicResultsPage<RowResult<Value>, byte[]>> result =
+                delegate.getFirstBatchForRanges(tableRef, rangeRequests, timestamp);
+        runWithExceptionHandling(() -> tracker.recordReadForTable(
+                tableRef, "getFirstBatchForRanges", MeasuringUtils.sizeOfPageByRangeRequestMap(result)));
+        return result;
+    }
+
+    @Override
+    public Set<TableReference> getAllTableNames() {
+        Set<TableReference> result = delegate.getAllTableNames();
+        runWithExceptionHandling(
+                () -> tracker.recordTableAgnosticRead("getAllTableNames", MeasuringUtils.sizeOf(result)));
+        return result;
+    }
+
+    @Override
+    public byte[] getMetadataForTable(TableReference tableRef) {
+        byte[] result = delegate.getMetadataForTable(tableRef);
+        runWithExceptionHandling(() -> tracker.recordTableAgnosticRead("getMetadataForTable", result.length));
+        return result;
+    }
+
+    @Override
+    public Map<TableReference, byte[]> getMetadataForTables() {
+        Map<TableReference, byte[]> result = delegate.getMetadataForTables();
+        runWithExceptionHandling(() -> tracker.recordTableAgnosticRead(
+                "getMetadataForTables", MeasuringUtils.sizeOfMeasurableByteMap(result)));
+        return result;
+    }
+
+    @Override
+    public Multimap<Cell, Long> getAllTimestamps(TableReference tableRef, Set<Cell> cells, long timestamp)
+            throws AtlasDbDependencyException {
+        Multimap<Cell, Long> result = delegate.getAllTimestamps(tableRef, cells, timestamp);
+        runWithExceptionHandling(
+                () -> tracker.recordReadForTable(tableRef, "getAllTimestamps", MeasuringUtils.sizeOf(result)));
+        return result;
+    }
+
+    @Override
+    public List<byte[]> getRowKeysInRange(TableReference tableRef, byte[] startRow, byte[] endRow, int maxResults) {
+        List<byte[]> result = delegate.getRowKeysInRange(tableRef, startRow, endRow, maxResults);
+        runWithExceptionHandling(() ->
+                tracker.recordReadForTable(tableRef, "getRowKeysInRange", MeasuringUtils.sizeOfByteCollection(result)));
+        return result;
+    }
+
+    private static void runWithExceptionHandling(Runnable trackingRunnable) {
+        try {
+            trackingRunnable.run();
+        } catch (Exception exception) {
+            log.warn("Key value service tracking runnable failed", exception);
+        }
+    }
+
+    private static <T> ClosableIterator<T> wrapIteratorWithExceptionHandling(
+            ClosableIterator<T> iterator, BytesReadTracker bytesReadTracker, ToLongFunction<T> measurer) {
+        ClosableIterator<T> result = iterator;
+        try {
+            result = new TrackingClosableIterator<>(iterator, bytesReadTracker, measurer);
```

Comment:
An idea here that is potentially worth exploring: maybe we add this to `KeyValueServiceDataTracker`, i.e., make a `KeyValueServiceDataTracker#wrapIterator` method (or maybe just `KeyValueServiceDataTracker#wrap`). This would also then help simplify the need for a `recordCallForTable`.

Reply:
Also I think we should be ok and not need to wrap this with an exception.